manifests/client/redhat.pp

```diff
@@ -1,58 +1,58 @@
 # Specific settings for client on Redhat distribution.
 class nfs::client::redhat inherits nfs::base {
 
-  package { 'nfs-utils':
-    ensure => present,
-  }
-
-  if versioncmp($::operatingsystemmajrelease, '6') == 0 {
-
-    package {'rpcbind':
-      ensure => present,
-    }
-
-    service {'rpcbind':
-      ensure    => running,
-      enable    => true,
-      hasstatus => true,
-      require   => [Package['rpcbind'], Package['nfs-utils']],
+  case  $::operatingsystemmajrelease {
+    '5' : {
+      $nfslock_requirement = [Package['nfs-client'], Package['nfs-utils']]
+      $nfsclient_package   = 'portmap'
+      $nfsclient_service   = 'portmap'
+      $nfslock_service     = 'nfslock'
+      $netfs_requirement  = [Service['nfs-client'], Service['nfslock']]
     }
-
-  } else {
-
-    package { 'portmap':
-      ensure => present,
+    '6' : {
+      $nfslock_requirement = [Service['nfs-client']]
+      $nfslock_service     = 'nfslock'
+      $nfsclient_service   = 'rpcbind'
+      $nfsclient_package   = 'rpcbind'
+      $netfs_requirement   = [Service['nfslock']]
     }
-
-    service { 'portmap':
-      ensure    => running,
-      enable    => true,
-      hasstatus => true,
-      require   => [Package['portmap'], Package['nfs-utils']],
+    default : {
```

Comment:
Should RH 7 be default really? There might be people still using RH4. I'd rather have default fail and list RH7 specificially.